### ISSUE TITLE
Crash on Startup, Beta version 1.2.0.0-325

### ISSUE THREAD

**Issue by Drakoflame** [SmithsModding/SmithsCore#9]
http://pastebin.com/wi31EYby

I know you tried to fix it Orion, but I think its crashing with another issue now, I can't tell whats causing it this time... I guess this is why we report bugs right? for the people who seems to be only able to get them XD

**Comment by marchermans** [comment]
Damn I might have missed a spot, could you do me a favor and find out which mod is causing this?

**Comment by marchermans** [comment]
And it would be nice if i can get the whole logf, and somehow acces to the pack, that way i can run it in the debug environment and see what item/block is causing the issue.

**Comment by marchermans** [comment]
So i did some additional Debugging and it seems that something is baking the models wrongly, I am going to need acces to those mods before i can help you.

By default Vanilla Forge bakes the Item/Block models with DefaultVertexFormats.ITEM which holds a normal that i have to reset. Yet in your instance something either additionaly bakes them with a different format (maybe optifine, i have not tried that, alltough there is no official Optifine release compatible with Forge for MC 1.10.x)  Or something modifies the way Bakedmodels are generated and changes the order which is VERY BAD and there is not much i can do then except for disabling it in that case.

I will see if i can make the normal reset function more dynamic, but i have not much hope with nothing to debug.

**Comment by marchermans** [comment]
I added some save quards to automatically detect if the VertexFormat has changed. Right now i am testing if that changed the default behaviour, i will be able to tell you more later.

**Comment by Drakoflame** [comment]
![modslist](https://cloud.githubusercontent.com/assets/8823913/16901677/18d3518a-4bff-11e6-8d8a-67f5136b63d5.png)
I've attached a screen cap of all current mods. All configs are default as far as I can remember, so you shouldn't have to worry about that. Let me know if there is anything else that you need help with.

OH! also, if it effects anything, I do use shaders, and if you need that one it is MrMeepz' Shaders v.05 LITE

**Comment by marchermans** [comment]
Try SmithsCore 1.10.2-1.2.0.0-201b version of SC and see if i fixed it.

**Comment by Drakoflame** [comment]
Was going to try it with the shader on and off, but didn't get a chance to before I crashed. Not sure if it was me holding down right click and opening the inventory, or placing the block. Same mods, no changes, no config changes, shaders were turned off, placed down a "forge" and crashed http://pastebin.com/xxS3P8S0

**Comment by marchermans** [comment]
Dann i missed a spot, this exception is known. Will fix it tomorrow.

**Comment by marchermans** [comment]
Oh and could zou zip your modsfolder and upload it somewhere, that way i have the exact same files.

**Comment by marchermans** [comment]
.And i pushed a sort of fix for that to Armory, hopefully it should be live in a couple of minutes

**Comment by Drakoflame** [comment]
from now on lets keep the discussions of chats to one topic, and not jump between the two, re-open this one, as it is where current and active logs are posted, I will close the other one after this post.

I crashed again shortly after putting down a forge http://pastebin.com/sdNcYqrZ

as simple as it would be to upload the files to dropbox, again, I have not changed anything, if your having any issues getting ahold of mods, which most likely are the Xaeros ones, they are easy to find, they're both available here http://chocolateminecraft.com/ other than that there should be no issues getting mods. No configs have been changed.

**Comment by marchermans** [comment]
A question do you run a vanilla MC 1.9.4 or MC 1.10.2 version of the game? Cause if you are running a 1.9.4 version this is caused by that. In 1.10 Mojang added the method getWorld to TileEntities and made worldObj private. As this is a official 1.10 version it is compiled against it and it uses this method. This would normally not cause any issues as FML remaps those calls to the obj on its own. Yet as the structure system is not based around TileEntities but around Istructureparts it does not get remapped. And the method does not exists on those TEs in 1.9 or below, causing the abstract method crash.

I will add the method myself and will upload a new version of School when I am home.

**Comment by Drakoflame** [comment]
[s]Alright, saw the latest release for SmithCore, downloaded it, using the latest version of Armory, and did my usual test, place a forge. I lasted... 2 seconds before the game locked up and "crashed" http://pastebin.com/QpvWXrVf tahts the errors that the output log on the launcher gave me... this is getting quite frustrating at this point, but I know I'm helping you guys iron out all the bugs...[/s]

Nevermind that, I just saw the update for Armory, downloaded that, placed a forge, and it worked :D I placed a forge and it didn't explode :D... I'm just curious... how does one make a blacksmiths anvil? when you require a blacksmiths anvil to make a blacksmiths anvil, a fireplace, a forge,a nd litterally everythng else... its impossible to start unless i'm looking in the wrong spot?

**Comment by Drakoflame** [comment]
p 3p alright... I'm back with a new report... I placed all placable blocks, the anvils were missing textures, as they have been. So, my first instinct was "can I make anything?" So I threw a hot iron ingot (that you get from the creative menu) into the stone blacksmiths anvil, and got this http://pastebin.com/iSWS5bBa

**Comment by marchermans** [comment]
On creating your first anvil:
1. place 27 Cobblestone in a 3x3x3 structure in your world!
2. right click one of the middleblocks in the side of the cube with your Smith's Hammer, provided it has enough durability it will create the Stone Anvil, you can progress from there.

And Drako you have not answered my question: Are you running a 1.9 or 1.10 instance?

**Comment by Drakoflame** [comment]
I'm using 1.10.2

**Comment by marchermans** [comment]
Me and my big mouth, i found the issue with your lastest reports, and it is entirely my fault.

I was using interfaces that had methods with that name, that way i did not need to make multiple of the same methods, but could just use one. Due to obfuscation (what minecraft does to cover its tracks) the vanilla methods get a different name, which i completly forgot, i donnot notice this in my dev environment cause it maps those different names back to their original ones when it loads mods, so i completly missed that.

But i fixed it now, i pushed a build for SC out and i am waiting on it to finish so i can grab it and update Armory for you, hopefully you can play then,

Greets,

Orion

**Comment by OrionDevelopment** [comment]
I tested it through with your mods and it somehow borks........

**Comment by Drakoflame** [comment]
ITS A MIRACLE :D It worked, I put an item in, and it didn't explode! (as well I made the stone anvil, love the model :D

I would like to report that the progress bar doesn't change, if I put a name up into the bar at the top, and hover over the progress bar it flickers between 0% and 2%. Also, I DID get an ingot to turn into nuggets, but that was after moving it all around the anvil gui, and at the time of writing, i'm having issues getting it to happen again.

Strike that last part, I had the item to hot, and it wasn't working, I did it again with another one at around 670 and the progress bar flickered and it turned into nuggets :D I'm going to have so much fun, just with the progress bar worked X3, maybe add a sound to show its working?

**Comment by Drakoflame** [comment]
also closing this since the issue has been "solved" opened a new one for another crash XD

**Comment by marchermans** [comment]
What do you mean with the progressbar does not change? Could you be a bit more specific, like send me a video or a gif of some sorts? Maybe hover over the arrow with your mouse?

**Comment by Drakoflame** [comment]
I have no idea how to make a gif... and the simplest way to explain it is, when I put my mouse over the arrow inside of the anvil GUI, a box that says "Progress" comes up next to the cursor, and it flickers between 0% and 2% if the temprature if close enough to the working temperature. Again, if there could also be a sound that plays occasionally, like a single strike of a hammer on the anvil, while something is being worked would be just as helpful. Also, if I put items into the gui that can be made into something, would it continue to make them if I closed out of the gui?

**Comment by marchermans** [comment]
Ah damn that is broken, i will open a proper issue for you under Armory

**Comment by marchermans** [comment]
I released both a updated SC and Armory version to fix that issue. The recipe for nuggets is shapeless as you can se now.
